### PR TITLE
Improved code readability and code quality

### PR DESCRIPTION
**Pull request by Sweetdevil144**
+ Fixed typos in several files to improve readability and grammar
+ Implemented methods to increase efficiency of algorithms

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.

### CODE REVIEW

**Comment by Sweetdevil144** [comment]
@BamaCharanChhandogi Successfully drafted the PR !. This is my second contribution to Algorithms/Java. The firstbeing prior to Hacktoberfest '23

**Review by debasishbsws — APPROVED** [review]
Nice work, Thanks

File: src/main/java/com/thealgorithms/maths/FindMinRecursion.java [hunk 1]
```diff
@@ -4,10 +4,21 @@ public final class FindMinRecursion {
...
+     * @param low the index of the first element
+     * @param high the index of the last element
+     * @return min of {@code array}
+     */
+
     public static int min(final int[] array, final int low, final int high) {
         if (array.length == 0) {
             throw new IllegalArgumentException("array must be non-empty.");
         }
+
```
Comment (debasishbsws) [inline src/main/java/com/thealgorithms/maths/FindMinRecursion.java:21]:
```suggestion

```

**Comment by Sweetdevil144** [comment]
Hello. I think the code is ready to be merged.

**Comment by Sweetdevil144** [comment]
@BamaCharanChhandogi

**Review by debasishbsws — APPROVED** [review]

**Pull request merged**